### PR TITLE
*: Improve rocksdb logger

### LINKED ISSUE #8117: raftdb log is printed into tikv log file
### What version of TiKV are you using?
master

### Steps to reproduce
Run tikv. Look at tikv log (e.g. deploy/log/tikv.log). Raftdb log are mixed with tikv log in the file.

After #7358 we switch to use tikv logger instead of rocksdb native logger to dump logs. By default rocksdb logs (kvdb and raftdb) should be in their respective directories (e.g. deploy/data/db and deploy/data/raft). On master, however, raftdb log is printed into tikv log. kvdb is fine though.

### PR DESCRIPTION
**Pull request by yiwu-arbug**
### What problem does this PR solve?

Issue Number: close #8117 

Problem Summary:

### What is changed and how it works?

What's Changed:
* Fix raftdb log printed into tikv log.
* Add `RocksFormater` to adjust rocksdb log format to closer to vanilla rocks logger.

### Related changes

- We changed to use tikv logger to for rocksdb log since #7358.
- The PR depends on https://github.com/tikv/rust-rocksdb/pull/518

### Check List 

Manual tested.

### Release note 

* No release note

### CODE REVIEW

File: components/engine_rocks/src/logger.rs [hunk 1]
```diff
@@ -27,7 +27,7 @@ impl Logger for RaftDBLogger {
         match log_level {
             InfoLogLevel::Debug => debug!(#"raftdb_log","{}", log),
             InfoLogLevel::Info | InfoLogLevel::Header | InfoLogLevel::NumInfoLog => {
-                info!(#"raft_log","{}", log)
+                info!(#"raftdb_log","{}", log)
```
Comment (yiwu-arbug) [inline components/engine_rocks/src/logger.rs:30]: This typo caused raftdb log printed into tikv log.

File: components/engine_rocks/src/logger.rs [hunk 2]
```diff
@@ -8,13 +8,13 @@ pub struct RocksdbLogger;
...
-            InfoLogLevel::Warn => warn!(#"rocksdb_log","{}", log),
-            InfoLogLevel::Error => error!(#"rocksdb_log","{}", log),
-            InfoLogLevel::Fatal => crit!(#"rocksdb_log","{}", log),
+            InfoLogLevel::Header => info!(#"rocksdb_log_header", "{}", log),
+            InfoLogLevel::Debug => debug!(#"rocksdb_log", "{}", log),
+            InfoLogLevel::Info => info!(#"rocksdb_log", "{}", log),
+            InfoLogLevel::Warn => warn!(#"rocksdb_log", "{}", log),
+            InfoLogLevel::Error => error!(#"rocksdb_log", "{}", log),
+            InfoLogLevel::Fatal => crit!(#"rocksdb_log", "{}", log),
+            _ => {}
```
Comment (Connor1996) [inline components/engine_rocks/src/logger.rs:17]:
```suggestion
            InfoLogLevel::NumInfoLog => {}
```
actually, I don't know what it is used for

File: components/engine_rocks/src/logger.rs [hunk 3]
```diff
@@ -25,13 +25,13 @@ pub struct RaftDBLogger;
...
-            InfoLogLevel::Warn => warn!(#"raftdb_log","{}", log),
-            InfoLogLevel::Error => error!(#"raftdb_log","{}", log),
-            InfoLogLevel::Fatal => crit!(#"raftdb_log","{}", log),
+            InfoLogLevel::Header => info!(#"raftdb_log_header", "{}", log),
+            InfoLogLevel::Debug => debug!(#"raftdb_log", "{}", log),
+            InfoLogLevel::Info => info!(#"raftdb_log", "{}", log),
+            InfoLogLevel::Warn => warn!(#"raftdb_log", "{}", log),
+            InfoLogLevel::Error => error!(#"raftdb_log", "{}", log),
+            InfoLogLevel::Fatal => crit!(#"raftdb_log", "{}", log),
+            _ => {}
```
Comment (Connor1996) [inline components/engine_rocks/src/logger.rs:34]: ditto

File: components/tikv_util/src/logger/mod.rs [hunk 4]
```diff
@@ -238,6 +260,57 @@ where
...
+                    chrono::Local::now().format(TIMESTAMP_FORMAT),
+                    thread::current().id().as_u64(),
+                )?;
+                decorator.start_level()?;
+                write!(decorator, "[{}]", get_unified_log_level(record.level()))?;
+                decorator.start_whitespace()?;
+                write!(decorator, " ")?;
+            }
+            decorator.start_msg()?;
+            let msg = format!("{}", record.msg());
```
Comment (Connor1996) [inline components/tikv_util/src/logger/mod.rs:302]: I suspect if here would introduce one more memory copy, maybe check `ends_with` directly on `record.msg()`?

File: components/engine_rocks/src/logger.rs (hunk 2 above)
Comment (yiwu-arbug) [reply components/engine_rocks/src/logger.rs:17]: The `NUM_INFO_LOG_LEVELS` is a c/c++ trick to get max enum number. In rust we do not need them. Let me update rust-rocksdb to remove this enum.

https://github.com/tikv/rocksdb/blob/6.4.tikv/include/rocksdb/env.h#L968

File: components/tikv_util/src/logger/mod.rs (hunk 4 above)
Comment (yiwu-arbug) [reply components/tikv_util/src/logger/mod.rs:302]: `record.msg()` is of type `fmt::Arguments` which cannot used as str.

**Comment by BusyJay** [comment]
It seems to be still WIP. Is it ready for review?

**Comment by yiwu-arbug** [comment]
> It seems to be still WIP. Is it ready for review?

yes, ready for review. WIP was only because https://github.com/tikv/rust-rocksdb/pull/518 haven't been merged.

**Pull request merged**